.github/workflows/run-all-tests.yml

```diff
@@ -19,6 +21,19 @@ jobs:
           sudo curl -O https://releases.hashicorp.com/vault/1.9.0/vault_1.9.0_linux_amd64.zip
           sudo unzip vault_1.9.0_linux_amd64.zip
           sudo mv vault /usr/local/bin
+      - name: Detect changes to Dockerfile
+        id: changed-dockerfile
+        uses: tj-actions/changed-files@v35.4.4
+        with:
+          files: Dockerfile
+      - name: Rebuild local image if Dockerfile changed
+        if: steps.changed-dockerfile.outputs.any_changed == 'true'
+        run: |
+          # the previous step uses the changed-files action to detect if the Dockerfile has any updates
+          # this will require rebuilding the image locally as the downstream "test" image will fail otherwise
+          # see https://github.com/marketplace/actions/changed-files for more info on this action
+          echo "#### CHANGES DETECTED TO DOCKERFILE, REBUILDING IMAGE LOCALLY ####"
+          docker build -t $DOCKER_IMAGE_NAME:development -f Dockerfile .
```

Comment:
Out-of-scope, but how much effort would it be to automatically `docker build` in local SCP via `docker-compose-setup.sh`?  It'd be convenient to not need to remember to manually `docker build` in cases of upstream change like this.

Reply:
That's a very interesting idea!  Let me play around with this for a minute... I actually think this is very doable.

Reply:
This does appear to be very straightforward:
```
LOCAL_BRANCH=$(git rev-parse --abbrev-ref HEAD)
CHANGED=$(git diff "$LOCAL_BRANCH" development --name-only -- Dockerfile)
```

This will compare your local branch with what the state of the Dockerfile is on `development`.  If there are any changes (staged or unstaged), then `$CHANGED` will be set to `Dockerfile`.  This can then be passed to the `if/else` to determine whether to build or pull the image.

Great suggestion!  I'm going to incorporate this and update the instructions.

Reply:
@eweitz [f2d272d](https://github.com/broadinstitute/single_cell_portal_core/pull/1713/commits/f2d272de450394c3cacbd511dca1a6f00bfb2648)